src/compiler/checker.ts

```diff
@@ -8856,8 +8856,18 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                     else {
                         const type = getWidenedType(getRegularTypeOfExpression(node.expression));
                         const computedPropertyNameType = typeToTypeNodeHelper(type, context);

```

Comment:
this can return `ImportType` in the enum's case, that's the reason why it crashed - the code here assumed that a literal type node would always be returned but that wasn't true